internal/tray-agent/agent/client/dashboard-client.go

```diff
@@ -0,0 +1,186 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"os"
+	"strings"
+)
+
+const (
+	//liqoDashboardIngressName is the name of the LiqoDash Ingress for remote access.
+	liqoDashboardIngressName = "liqo-dashboard-content"
+	//liqoDashboardServiceName is the name of the LiqoDash Service.
+	liqoDashboardServiceName = "liqo-dashboard"
+	//liqoDashboardNamespace is the name of the LiqoDash namespace.
+	liqoDashboardNamespace = "liqo"
+	//liqoDashboardSAName is the name of the LiqoDash ServiceAccount containing the reference
+	//to the secret with the access token.
+	liqoDashboardSAName = "liqodash-admin-sa"
+	//liqoDashboardTkPrefix is the name prefix of the LiqoDash access token.
+	liqoDashboardTkPrefix = "liqodash-admin-sa-token"
+	//masterNodeLabel is the label name associated with master nodes.
+	masterNodeLabel = "node-role.kubernetes.io/master"
+	//EnvLiqoDashHost defines the env var for the HOST part of the LiqoDash address.
+	EnvLiqoDashHost = "LIQODASH_HOST"
+	//EnvLiqoDashPort defines the env var for the PORT part of the LiqoDash address.
+	EnvLiqoDashPort = "LIQODASH_PORT"
+)
+
+//AcquireDashboardConfig tries to retrieve required data to access the LiqoDash service.
+//
+//If a valid configuration is found, the EnvLiqoDashHost and EnvLiqoDashPort env vars are set.
+func (ctrl *AgentController) AcquireDashboardConfig() error {
+	//cleanup LiqoDash env vars
+	var err error
+	if err = os.Unsetenv(EnvLiqoDashHost); err != nil {
+		return err
+	}
+	if err = os.Unsetenv(EnvLiqoDashPort); err != nil {
+		return err
+	}
+	//preliminary check to verify the LiqoDash pod is running
+	var dashPodL *corev1.PodList
+	dashPodL, err = ctrl.kubeClient.CoreV1().Pods(liqoDashboardNamespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "app=" + liqoDashboardServiceName,
+		FieldSelector: fields.OneTermEqualSelector("status.phase", "Running").String(),
```

Comment:
Suggestion: May the check be hardened looking if the pod is ready to serve traffic and not only running?